packages/grafana-runtime/src/services/SidecarService_EXPERIMENTAL.ts

```diff
@@ -20,14 +30,35 @@ import { HistoryWrapper, LocationService, locationService } from './LocationServ
 export class SidecarService_EXPERIMENTAL {
   private _initialContext: BehaviorSubject<unknown | undefined>;
   private memoryLocationService: LocationService;
-  private history: LocationStorageHistory;
+  private mainLocationService: LocationService;
+  private follow = false;
+  private mainLocationWhenOpened: string | undefined;
+  private mainOnAllowedRoute = false;
 
-  constructor() {
+  constructor(mainLocationService: LocationService) {
     this._initialContext = new BehaviorSubject<unknown | undefined>(undefined);
-    // We need a local ref for this so we can tap into the location changes and drive rerendering of components based
-    // on it without having a parent Router.
-    this.history = createLocationStorageHistory({ storageKey: 'grafana.sidecar.history' });
-    this.memoryLocationService = new HistoryWrapper(this.history);
+    this.mainLocationService = mainLocationService;
+    this.memoryLocationService = new HistoryWrapper(
+      createLocationStorageHistory({ storageKey: 'grafana.sidecar.history' })
+    );
+
+    this.mainOnAllowedRoute = ALLOW_ROUTES.some((prefix) =>
+      mainLocationService.getLocation().pathname.startsWith(prefix)
+    );
+
+    this.mainLocationService.getLocationObservable().subscribe((location) => {
+      this.mainOnAllowedRoute = ALLOW_ROUTES.some((prefix) => location.pathname.startsWith(prefix));
```

Comment:
So this whole thing is basically a memory leak in the sense that there is no unsubscribe, but not sure what that would be called and for most part I think both locationServices and this service will be created just once per grafana lifetime so should not be a big deal. Open for suggestions thought because I don't like it as a principle.